src/commands/moderation/clear.ts

```diff
@@ -0,0 +1,78 @@
+import { ApplicationCommandOptionType, ApplicationCommandType, TextChannel, Colors } from 'discord.js';
+import { slashCommand, slashCommandStructure, makeEmbed, constantsConfig } from '../../lib';
+
+const data = slashCommandStructure({
+    name: 'clear',
+    description: 'Delete a specified number of messages in the current channel.',
+    type: ApplicationCommandType.ChatInput,
+    default_member_permissions: constantsConfig.commandPermission.MANAGE_SERVER,
+    dm_permission: false,
+    options: [{
+        name: 'amount',
+        description: 'Number of messages to delete (1-100).',
+        type: ApplicationCommandOptionType.Integer,
+        required: true,
+        min_value: 1,
+        max_value: 100,
+    }],
+});
+
+export default slashCommand(data, async ({ interaction, log, client }) => {
+    const amount = interaction.options.getInteger('amount');
+
+    if (!interaction.channel || !interaction.channel.isTextBased()) {
+        return interaction.reply({ content: 'This command can only be used in text channels.', ephemeral: true });
+    }
+
+    if (amount === null) {
```

Comment:
Why not check if amount is `falsy` instead of specifically checking against `null`?
```suggestion
    if (!amount) {
```